### DISCUSSION TITLE
We need to update the guide

### DISCUSSION THREAD

**Discussion by teril43** [d8ahazard/sd_dreambooth_extension#378] [Q&A]
I don't know how to use the separated concepts tab after changing to the new ui.

always grateful for your efforts.

**Comment by FugueSegue** [comment]
I just got the Dreambooth extension working again and I was about to train a new model and I understand the settings less than I did before. If the guide is updated, there needs to be examples provided. Don't assume that anyone has done any Dreambooth training before.

**Comment by d8ahazard** [comment]
I'm going to go on a mild rant here. It's a general rant, so don't take it personally. That said...


You guys realize that you can hover over the majority of UI elements to see hint text indicating what things do, right?

![image](https://user-images.githubusercontent.com/1633844/204915715-b4043b44-e46f-4098-9d01-580f5d738944.png)


You also realize that I'm just one guy, right? This is what my inbox has looked like every day for the past month or so:

![image](https://user-images.githubusercontent.com/1633844/204915594-fadde074-15d4-4d3a-a7d9-2077e01dfc56.png)


If you really want a good time, go through the issues and look at *just* the closed ones. Then look at the list of pending features. Then, for shits and giggles, look at ALL of the issues people submit without using the fu*$ing template, or who don't bother to update the UI and plugin and restart like the fu*$ing template specifically requests people do.

The sheer amount of administrative B$ required just to be able to keep up with relevant issues because people can't bother to read is staggering.


All that said - let me circle back to - "I'm just one guy.". @sgsdxzy  has been helping out a bit, but it's still my can of worms. Every day I wake up and I'm like "Hmmm, should I try to add 2.0 support today?" or maybe "Maybe today is the day where I can find some time to try to implement aspect-ratio bucketing."

Nope. I spend an hour each morning just reading over issues, answering the ones I can, and closing the ones where people were too danged lazy to follow a simple form.


And, of course, I still need to test whatever I do wind up working on, which is also usually very time-consuming. 

THEN - if I wind up at a point in my day where I'm satisfied the code appears to be not broken and everything I test is working and the damned thing can just train...you know what? Some times, I like to actually *use* Stable-Diffusion and Dreambooth, and not just add more features to it. Call it trying to enjoy the fruits of my labor.


On top of all this, I do have a normal job, and three kids, and a wife, and a house to take care of and a goddamned dog that nobody but me seems to be able to let outside...and my god, is it already 9PM? Did I really just spend another 12 hours working almost exclusively on this? Jesus...


So, what I'm saying is - I'm doing the best I can. I've got this great idea to clone myself a few times to be able to double or triple what I can accomplish in a day, but that's pretty expensive and I don't think another miniature version of me who's not even housebroken is going to do a lot of programming...

As such, I'm always open to Pull requests to update the documentation, volunteers to help update/edit/build the Wiki, or even someone to say "Hey, UI element 375/1000 doesn't have a hint."

Hell, if you wanna update the readme, let me know, and I'll let you put me on speed dial so you can pick my brain.


But, at least for the present moment, I just don't have enough time in my day.

**Reply by chrisparkernz** [reply]
Legend - it's appreciated!

**Reply by Derjyn** [reply]
Keep it up, dude. You've created a valuable asset in the scene, and while it may or may not be apparent to you yet, you've been etched into the stone monolith because of it.

Take care of you and yours first, and feel no pain in making _this_ wait. The worst thing you can do to yourself and this passion, is start to loathe it. I left a salary job "in the industry" to work for barely-above-minimum wage at a smoke shop in the slums of my city. Why? Because the "dream" job resulted in my soul being left somewhere at a truck stop and I stopped enjoying what I used to love. Now I can still pay my bills (barely), my soul is back on radar, and I can get back to being a nerd and enjoying it. I have time for others.

Take care of yourself, take care of your family, and please for the love of Baphomet... Pick up the poops your doggo leaves as presents. Doggo will appreciate it, and so will humanity. After all that, if you haven't put yourself into a food coma, and you have the fire to do so, chip away at the DreamBooth extension. We'll all understand. Those who don't, can just chew on a bag of that doggo present.

**Reply by teril43** [reply]
I'm sorry.

I know there is a hint text, but I think it is my fault that I do not understand it alone.

Still, I haven't solved the problem, but I understand what you're saying because I'm in a similar position.

I hope you don't feel so stressed about this.
Thank you for your hard work.

**Reply by ClashSAN** [reply]
I joined github solely to help answer issues on the webui page, which were being spammed pretty badly and with many duplicate issues. I'd love to help ease any burden you have so you can work on features instead (what we all want)

but, I'll have to do alot of issue reading first. And I don't actually have enough vram to run this, so either I experiment on cpu or try to use google colab for this. 

(So, guess I probably can't be helpful until I play around with this locally with a better gpu)

**Comment by FugueSegue** [comment]
I appreciate the time you're putting into this. So do many others. I sympathize.

I am absolutely dedicated to mastering SD. I spend my days and nights processing images and experimenting. And I'm trying to take careful notes. Once I figure it out, I'll be glad to volunteer to help write/update documentation.

This is important to me. A.I. is the most significant thing to happen in the field of computer art since an anonymous IBM employee generated an image of a pin-up girl on a military radar screen back in 1956. That was the first documented work of computer art. Yes, weebs have been using computers to make pictures of their waifu since day one.

I've been fascinated with pixels on the video screen since I saw a demo of Pong at a Sears showroom when I was a toddler. My first drawing tablet was for my Atari 800XL. I had to sign up for access to Photoshop version 1 at the local college's computer lab. I earned a BFA Computer Art at SCAD when that was the only place in the world that offered that degree.

These days I program custom Python image processors and acrylic paint mixtures for my unique style of hand-painted pixel art--a style that, incidentally, probably can't been trained into an SD model. This fantastic way of generating images is exactly what I need right now. My life is literally on hold until I can define a new workflow that utilizes A.I. art.

Once more people get a grasp of how this stuff works, it's going to get easier. Because me and others will help write something that resembles a manual. Yes, I noticed all of the hints that pop up when I hover over all the UI elements. And I've been taking notes of what they say and then I search any place I can on the internet to better understand. I spent all of today trying to wrap my head around the Concepts tab and what all the new parameters mean. Because I'm so new at using Dreambooth, I barely understood the old parameters. And for what it's worth, I like what you've recently done to improve it.

I hate asking noob questions on the internet because I learned the valuable lesson of "reading the f*cking manual" a long time ago. But SD and Dreambooth do not have manuals. Just scattered Reddit posts, miscellaneous blogs, incomprehensible scientific papers, histrionic news articles, and YouTube videos that all become outdated within days because of new advancements and updates. I try my best to carefully research before I ask questions. But in this case it's very difficult. I'm doing my best.

SD and Dreambooth are so new that there's barely anything to reference. Your time is better spent programming. Once more people understand how all this works, you won't be dealing with newbie questions all the time.

I also want to thank you for helping me earlier today. Adding --ckptfix to the COMMANDLINE_ARGS did the trick and I was able to build a model. I thus spent the rest of the day trying to understand everything about classification prompts and so on. By suppertime I felt confident that I had set up a proper training. It generated lots of class images. And then I got another CUDA OOM. I'll figure it out.

Best of luck and hang in there.

**Comment by FugueSegue** [comment]
I am working on a draft for a Dreambooth extension guide. Is that okay with you, d8ahazard?
